spec/index.bs

```diff
@@ -730,51 +730,47 @@ For each [=subresource=] of a [=physical resource=], its set of
+- Outside of a pass (in {{GPUCommandEncoder}}), each (non-state-setting) command is one usage scope
     (e.g. {{GPUCommandEncoder/copyBufferToTexture()}}).
-- In a compute pass, each command is one usage scope.
+- In a compute pass, each non-state-setting command is one usage scope.
     A subresource is "used" in the usage scope if it's accessible by the command.
     For example, in {{GPUComputePassEncoder/dispatch()}},
     every subresource in every currently bound bind group is "used" in the usage scope.
-    (Compute pass state-setting commands are trivially-valid usage scopes; see below.)
 - One render pass is one usage scope.
-    A subresource is "used" in the usage scope if it's referenced by any command,
-    including state setting commands
-    (like {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)}}).
+    A subresource is "used" in the usage scope if it's referenced by any
+    (state-setting or non-state-setting) command. For example, in
+    {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)}},
+    every subresource in `bindGroup` is "used" in the render pass's usage scope.
 
 Issue: The above should probably talk about [=GPU commands=]. But we don't have a way to
 reference specific GPU commands (like dispatch) yet.
 
-Resources that are used in state-setting calls in passes are always added as used resources in the containing [=usage scope=].
-
 <div class=note>
-    This means the following example resource usages **are** included in [=usage scope validation=]:
-
-    - Resources used in every {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)|GPUProgrammablePassEncoder.setBindGroup}}
-        regardless of whether the shader or {{GPUPipelineLayout}} actually depends on these bindings, or the bind group is shadowed by another 'set' call.
-    - Vertex buffer used in every {{GPURenderEncoderBase/setVertexBuffer()|GPURenderEncoderBase.setVertexBuffer}}
-        regardless of whether draw call depends on this buffer, or this buffer is shadowed by another 'set' call.
-    - Index buffer used in every {{GPURenderEncoderBase/setIndexBuffer()|GPURenderEncoderBase.setIndexBuffer}}
-        regardless of whether draw call depends on this buffer, or this buffer is shadowed by another 'set' call.
-    - Textures used as color attachments, resolve attachments or depth/stencil attachment in {{GPURenderPassDescriptor}} by {{GPUCommandEncoder/beginRenderPass()|GPUCommandEncoder.beginRenderPass()}}
+    The above rules mean the following example resource usages **are** included in [=usage scope validation=]:
+
+    - Subresources used in any
+        {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)|setBindGroup()}}
+        call, regardless of whether the currently bound pipeline's
+        shader or layout actually depends on these bindings,
+        or the bind group is shadowed by another 'set' call.
```

Comment:
Do we need to separate "shadowed bind group" and say resources in shadowed bind group should be validated in render pass (but not compute pass)?  Note that index buffer and vertex buffer are not included in compute pass, so there are no shadowed vertex buffer or index buffer in compute at all.

Reply:
Fixed by making this example specifically for render passes. It's already explained above that state-setting compute pass command don't contribute directly to a usage scope.